From a9c68ab977d0245a72ef467395c1dcb32e4928a3 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 12 Apr 2016 12:30:24 -0400
Subject: [PATCH 1011/1110] drm/amd/dal: make a type-safe cgs_device struct.

This is just a type-safety things to avoid everyone taking void *,
it doesn't change anything.

agd5f: split out from Dave's original patch that included
base amdgpu and dal.

Signed-off-by: Dave Airlie <airlied@redhat.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.h | 2 +-
 drivers/gpu/drm/amd/dal/dc/dc.h               | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.h b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.h
index 34f1f9f..d497b6d 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.h
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.h
@@ -75,7 +75,7 @@ struct irq_list_head {
 struct amdgpu_display_manager {
 	struct dal *dal;
 	struct dc *dc;
-	void *cgs_device;
+	struct cgs_device *cgs_device;
 	/* lock to be used when DAL is called from SYNC IRQ context */
 	spinlock_t dal_lock;
 
diff --git a/drivers/gpu/drm/amd/dal/dc/dc.h b/drivers/gpu/drm/amd/dal/dc/dc.h
index d5bb183..b8d0786 100644
--- a/drivers/gpu/drm/amd/dal/dc/dc.h
+++ b/drivers/gpu/drm/amd/dal/dc/dc.h
@@ -64,7 +64,7 @@ struct dc_init_data {
 	struct hw_asic_id asic_id;
 	struct dal_override_parameters display_param;
 	void *driver; /* ctx */
-	void *cgs_device;
+	struct cgs_device *cgs_device;
 
 	uint8_t num_virtual_links;
 	/*
-- 
2.7.4

